x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -813,56 +877,52 @@ public void clusterChanged(ClusterChangedEvent event) {
 
-    static class LocalNodeInfo {
+    static class EntryCommonFields {
         private final Settings settings;
         private final DiscoveryNode localNode;
-        final String prefix;
-        private final String localOriginTag;
+        final Map<String, String> commonFields;
 
-        LocalNodeInfo(Settings settings, @Nullable DiscoveryNode newLocalNode) {
+        EntryCommonFields(Settings settings, @Nullable DiscoveryNode newLocalNode) {
             this.settings = settings;
             this.localNode = newLocalNode;
-            this.prefix = resolvePrefix(settings, newLocalNode);
-            this.localOriginTag = localOriginTag(newLocalNode);
-        }
-
-        static String resolvePrefix(Settings settings, @Nullable DiscoveryNode localNode) {
-            final StringBuilder builder = new StringBuilder();
-            if (HOST_ADDRESS_SETTING.get(settings)) {
-                final String address = localNode != null ? localNode.getHostAddress() : null;
-                if (address != null) {
-                    builder.append("[").append(address).append("] ");
+            final MapBuilder<String, String> commonFieldsBuilder = new MapBuilder<>();
+            commonFieldsBuilder.put(EVENT_CATEGORY_FIELD_NAME, "elasticsearch-audit");
+            if (EMIT_NODE_NAME_SETTING.get(settings)) {
+                final String nodeName = Node.NODE_NAME_SETTING.get(settings);
+                if (Strings.hasLength(nodeName)) {
+                    commonFieldsBuilder.put(NODE_NAME_FIELD_NAME, nodeName);
                 }
             }
-            if (HOST_NAME_SETTING.get(settings)) {
-                final String hostName = localNode != null ? localNode.getHostName() : null;
-                if (hostName != null) {
-                    builder.append("[").append(hostName).append("] ");
+            if (newLocalNode != null) {
+                if (EMIT_HOST_ADDRESS_SETTING.get(settings) && Strings.hasLength(newLocalNode.getHostAddress())) {
+                    commonFieldsBuilder.put(HOST_ADDRESS_FIELD_NAME, newLocalNode.getHostAddress());
                 }
-            }
-            if (NODE_NAME_SETTING.get(settings)) {
-                final String name = Node.NODE_NAME_SETTING.get(settings);
-                if (name != null) {
-                    builder.append("[").append(name).append("] ");
+                if (EMIT_HOST_NAME_SETTING.get(settings) && Strings.hasLength(newLocalNode.getHostName())) {
+                    commonFieldsBuilder.put(HOST_NAME_FIELD_NAME, newLocalNode.getHostName());
                 }
             }
-            return builder.toString();
+            // the default origin is local
+            commonFieldsBuilder.put(ORIGIN_TYPE_FIELD_NAME, LOCAL_ORIGIN_FIELD_VALUE);
+            if (newLocalNode != null) {
+                commonFieldsBuilder.put(ORIGIN_ADDRESS_FIELD_NAME, newLocalNode.getHostAddress());
+            }
+            this.commonFields = commonFieldsBuilder.immutableMap();
```

Comment:
common fields include: `node name`, `host name`, `host ip` and, implicitly originate locally.